cmd/client/main.go

```diff
@@ -55,21 +64,28 @@ type flags struct {
 }
 
 func main() {
-	flags := &flags{}
+	flagset := &flags{}
 	rootCmd := &cobra.Command{
 		Use:   "client",
 		Short: "Starts a grpc or connect client, based on implementation",
 		Run: func(cmd *cobra.Command, args []string) {
-			run(flags)
+			run(flagset)
 		},
 	}
-	rootCmd.Flags().StringVar(&flags.host, "host", "127.0.0.1", "the host name of the test server")
-	rootCmd.Flags().StringVar(&flags.port, "port", "", "the port of the test server")
-	rootCmd.Flags().StringVarP(
+	if err := bind(rootCmd, flagset); err != nil {
+		os.Exit(1)
+	}
+	_ = rootCmd.Execute()
+}
+
+func bind(cmd *cobra.Command, flags *flags) error {
+	cmd.Flags().StringVar(&flags.host, hostFlagName, "127.0.0.1", "the host name of the test server")
+	cmd.Flags().StringVar(&flags.port, portFlagName, "", "the port of the test server")
+	cmd.Flags().StringVarP(
 		&flags.implementation,
-		"implementation",
+		implementationFlagName,
 		"i",
-		"connect",
+		"",
```

Comment:
remove default value as it is outdated and I cannot think of a good default, marked it as required instead